proc_macros/src/rule.rs

```diff
@@ -393,6 +400,16 @@ fn get_rule_rule_impl(
                 Some(capture_name) => quote!(Some(#capture_name.into())),
                 None => quote!(None),
             });
+    let maybe_deserialize_options = match rule.options_type.as_ref() {
+        None => quote!(),
+        Some(options_type) => {
+            quote! {
+                let options: #options_type = options.map(|options| {
+                    #crate_name::serde_json::from_str(&options.to_string()).expect("Couldn't parse rule options")
```

Comment:
Ok so this is where we're "strongly typing" the rule-specific options that got passed

Prior to this it "generically parsed" any `options` that it saw in the config file contents into `serde_json::Value`'s (which it's nice/interesting that you can parse YAML into)

So... I don't know if you can "deserialize" "straight" from `serde_json::Value` -> some specific `Deserialize` type?

But so what I did is re-serialize (into JSON) and then re-deserialize into the known rule-declared specific type